py-polars/polars/series/series.py

```diff
@@ -481,14 +481,30 @@ def _comp(self, other: Any, op: ComparisonOperator) -> Series:
                 return self.clone()
             elif (other is False and op == "eq") or (other is True and op == "neq"):
                 return ~self
-
-        if isinstance(other, datetime) and self.dtype == Datetime:
-            time_zone = self.dtype.time_zone  # type: ignore[union-attr]
-            if str(other.tzinfo) != str(time_zone):
-                raise TypeError(
-                    f"Datetime time zone {other.tzinfo!r} does not match Series timezone {time_zone!r}"
+        elif isinstance(other, float) and self.dtype in INTEGER_DTYPES:
```

Comment:
Casting from int -> float can be dangerous if the ints are large, should we do any checks or warnings here?

Reply:
This is hard to solve. We have the same problem in our supertype conversion and don't have a good answer to them.